src/types.ts

```diff
@@ -2238,3 +2250,10 @@ export type TaskStatus = {
   };
   result?: UnknownType;
 };
+
+export type TruncateOptions<AttachmentType, MessageType, UserType> = {
+  hard_delete?: boolean;
+  message?: Message<AttachmentType, MessageType, UserType>;
```

Comment:
let's pass in all the generics here:
   AttachmentType,
    ChannelType,
    CommandType,
    MessageType,
    ReactionType,
    UserType